validphys2/src/validphys/commondataparser.py

```diff
@@ -41,6 +67,363 @@
 }
 
 
+@Parser
+def ValidPath(path_str: str) -> Path:
+    """Parse strings into paths"""
+    return Path(path_str)
+
+
+@Parser
```

Comment:
If we don't care about supporting random-case stuff, we can as well have `Literal[*OP]` instead of this.

Reply:
You mean you want to support them or you don't? I don't mind either way tbh.

Reply:
I think we can make `ValidOperation = typing.Literal[*OP]`. The last version  is incorrect in that it would break if you pass a non sting (when calling `upper`)

Reply:
At the moment OP is in another module and would create circular dependencies. I've casted the str (and I've also covered the case `None` since the `null` as entering `None`). We can decide at the end whether to support
 random-case or not. Same for `ValidPath` above, might get converted to Path with no checks.

Reply:
Then can we not copy the dict keys (or even move that dict and related to a separate module?)

Why do we want to support None at all? Can we not make it "NULL" only?

Also the latest version will sill crash for e.g. `op_str = ["patata"]`; we really want to have an annotation for the input argument (which could be `str`, and you could have that the type is `Optional[ValidOperation]`).

Reply:
My idea is to move the operations when I'm done with the parser. They should not be part of `convolutions` anyway, but of the coredata or parser. But I want to limit changes to other parts of the code until I'm done so that I can rebase on top of master without suffering too much.

> Why do we want to support None at all?

Because `yaml.safe_load("null")` is `None` so I'm making it "NULL" when it comes as `None` since it's likely that people will forget the " when writing the operations. 

Maybe we want to be very strict but I just tried with your suggestion and it doesn't work but even if it worked, I think an error msg which will say something like `you tried to use None, accepted values are "NULL"` when they have written `operation: null` is only going to annoy people.

Reply:
> My idea is to move the operations when I'm done with the parser. They should not be part of `convolutions` anyway, but of the coredata or parser. But I want to limit changes to other parts of the code until I'm done so that I can rebase on top of master without suffering too much.

I'd suggest just copying the keys then. For one, it leads to better error messages.

Reply:
You mean now or as the final version?

> Personally, I could live with that on my conscience.

I don't know. Given that it is a mistake so easy to anticipate it seems a bit mean :__ Also, it's on us for choosing "NULL" as the name of the operation I guess.

Reply:
> You mean now or as the final version?

Not sure what the question is referring to, but if it is the aliasing thing, then definitively yes, before there are more of these files and conflicting versions of the format. For using Literal, it doesn't matter that much, but it would be a mild improvement for some duplication.

Reply:
Just to make sure I understand the proposal:

Change all new commondata files to use `ID` instead of `NULL`.
For old theories, since they will have `NULL` in their compound files, change it directly to `ID` upon reading.

Reply:
The proposal would be

```diff
diff --git a/validphys2/src/validphys/convolution.py b/validphys2/src/validphys/convolution.py
index 94e4c7fcd..3b54b26db 100644
--- a/validphys2/src/validphys/convolution.py
+++ b/validphys2/src/validphys/convolution.py
@@ -93,6 +93,7 @@ OP = {
     "COM": _com,
     "SMT": _smt,
     "NULL": _id,
+    "ID": _id,
 }
 
 class PredictionsRequireCutsError(Exception): pass
```

and the banning "NULL" from new files.